packages/jsapi-components/src/useBroadcastChannel.ts

```diff
@@ -0,0 +1,39 @@
+import {
+  BROADCAST_CHANNEL_NAME,
+  isMessage,
+  Message,
+} from '@deephaven/jsapi-utils';
+import Log from '@deephaven/log';
+import { EMPTY_FUNCTION } from '@deephaven/utils';
+import { useEffect, useMemo } from 'react';
+
+const log = Log.module('useBroadcastChannel');
+
+export function useBroadcastChannel(
+  onEvent: (event: MessageEvent<Message<unknown>>) => void = EMPTY_FUNCTION,
+  name = BROADCAST_CHANNEL_NAME
+) {
+  const channel = useMemo(() => new BroadcastChannel(name), [name]);
+  useEffect(
+    () => () => {
+      channel.close();
+    },
+    [channel]
+  );
+
+  useEffect(() => {
+    channel.onmessage = (event: MessageEvent) => {
```

Comment:
You can also do `channel.addEventListener('message', (event) => {})`. They co-exist w/ the `onmessage` it seems although you can only set `onmessage` to 1 handler

Maybe add a comment here that (I'm assuming) this is intentionally **NOT** `addEventListener` so that you can set the new listener without needing to remove the old listener. Or change to cleanup the event listener from the effect if you switch to `addEventListener`

Reply:
Yes, `addEventListener` does not do what you would expect and would be wrong. I have a comment about this in `MessageUtils`, but I'll move that comment over here instead as it's more relevant here:
```
/**
 * Use a BroadcastChannel for sending messages between tabs, such as when the user logs out.
 * Something that isn't clear from the docs - for inter-browser communication, you must use `postMessage` and `onmessage`.
 * `BroadcastChannel` is an `EventTarget`, and does support `addEventListener`/`dispatchEvent` etc... but that's only within the same instance of that object!
 * https://developer.mozilla.org/en-US/docs/Web/API/Broadcast_Channel_API
 */
```

Reply:
I tried `addEventListener` with `postMessage` and it seemed to work. Just FYI

Reply:
Weird. I wonder why I previously made that comment, as I can find no evidence now that addEventListener ever behaved differently. Will just use `addEventListener` and remove the comment.